src/Groups/matrices/iso_nf_fq.jl

```diff
@@ -0,0 +1,172 @@
+      return false, Fp, matrices_Fp
+   end
+
+   for M in matrices
+      for i = 1:nrows(M)
+         for j = 1:ncols(M)
+            if iszero(M[i, j])
+               continue
+            end
+
+            if mod(denominator(M[i, j]), p) == 0
+               return false, Fp, matrices_Fp
+            end
+         end
+      end
+   end
+   # I don't want to invert everything in char 0, so I just check whether the
+   # matrices are still invertible mod p.
+   for i = 1:length(matrices)
+      matrices_Fp[i] = map_entries(Fp, matrices[i])
+      if rank(matrices_Fp[i]) != nrows(matrices_Fp[i])
+         return false, Fp, matrices_Fp
+      end
+   end
+
+   return true, Fp, matrices_Fp
+end
+
+# Detinko, Flannery, O'Brien "Recognizing finite matrix  groups over infinite
+# fields", Section 3.2 claims that any prime != 2 not dividing the discriminant
+# of the defining polynomial and not dividing any denominator of the matrices
+# and their inverses (!) works.
+function test_modulus(matrices::Vector{T}, p::Int) where T <: AbstractAlgebra.MatElem{nf_elem}
+   @assert length(matrices) != 0
+   K = base_ring(matrices[1])
+   matrices_Fq = Vector{fq_mat}(undef, length(matrices))
+   if p == 2
+      return false, FiniteField(fmpz(p), 1, "a")[1], matrices_Fq
+   end
+   O = EquationOrder(K)
+   if mod(discriminant(O), p) == 0
+      return false, FiniteField(fmpz(p), 1, "a")[1], matrices_Fq
+   end
+   for M in matrices
+      for i = 1:nrows(M)
+         for j = 1:ncols(M)
+            if iszero(M[i, j])
+               continue
+            end
+
+            if mod(denominator(M[i, j]), p) == 0
+               return false, FiniteField(fmpz(p), 1, "a")[1], matrices_Fq
+            end
+         end
+      end
+   end
+
+   # p is does not divide disc(O), so it's not an index divisor, so we don't
+   # have to work in the maximal order here.
+   P = prime_ideals_over(O, p)
+   Fq, OtoFq = ResidueField(O, P[1])
+   # I don't want to invert everything in char 0, so I just check whether the
+   # matrices are still invertible mod p.
+   for i = 1:length(matrices)
+      matrices_Fq[i] = matrix(Fq, [ OtoFq(O(numerator(a)))//OtoFq(O(denominator(a))) for a in matrices[i] ])
+      if rank(matrices_Fq[i]) != nrows(matrices_Fq[i])
+         return false, Fq, matrices_Fq
+      end
+   end
+
+   return true, Fq, matrices_Fq
+end
+
+# Returns the maximal possible order of a finite subgroup of GL(n, QQ).
+# Always return a fmpz, only the orders for n <= 16 would fit into an Int64.
+# This relies on results in a preprint "The orders of finite linear groups" by
+# W. Feit (1995), possibly published as mathscinet.ams.org/mathscinet-getitem?mr=1484185
+# in the "Proceedings of the First Jamaican Conference on Group Theory and its
+# Applications", 1996. However, I can't find either paper.
+# Geoff Robinson claims to have the preprint and posted the relevant information
+# at mathoverflow.net/questions/168292/maximal-order-of-finite-subgroups-of-gln-z .
```

Comment:
I agree. In Hecke there is `_multiple_minkowski` which returns a common multiple of the orders of all subgroups of GL(n, QQ) given QQ (assuming I got that right, @thofma knows more). I think that one is published. We could also use this (and check divisibility instead of < in line 16).

Reply:
I dug and found a better reference, see https://mathoverflow.net/questions/168292/maximal-order-of-finite-subgroups-of-gln-z/400599#400599

I will push a commit to update this.

Reply:
And I guess the comments in Hecke.jl before `_multiple_of_finite_group_order_glnz` resp. `_minkowski_multiple` could also be updated based on that information (CC @thofma).

Reply:
Actually, scratch that, I misread that paper -- it *quotes* Feit, it does not reprove his results :-(.

So the closest I could find to a "proof" of Feit's results is his paper https://doi.org/10.1090/S0002-9939-97-03801-X where he proves relevant results, but never explicitly "shows the numbers". I tried reimplementing his formulas, and I get bounds close to the ones we have here, but not quite equal (they are multiples of his bounds from the Jamaican conference).

This is *extremely* unsatisfying, I must say. Perhaps time to assign a student to redo this classification *sigh*